### DISCUSSION TITLE
Connect Mealie to 3rd Party Services Like Bring Shopping List, Google Assistant, Alexa etc.

### DISCUSSION THREAD

**Discussion by crow1093** [mealie-recipes/mealie#2589] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Maybe it's possible to sync the Shopping List to Bring or have Recipes on Smart Displays

### Please provide a concise description of the feature that would resolve your issue.

.

### Please consider and list out some caveats or tradeoffs made in your design decision

.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
I wrote an app to sync Mealie's shopping list to/from Alexa:

https://shopping-lists.genson.dev/

I wanted to build it into Mealie directly, but it requires a central authentication service, which fundamentally doesn't work with a self hosted app.

I'm open to adding support for other services as requested. Unfortunately it's impossible to integrate with Google Assistant because there's no API (which blows my mind). I think I looked into Bring and it doesn't support real time notifications, which makes a bidirectional sync impossible, but I'd be happy to look again if there's enough interest

**Reply by ulimanaio** [reply]
Bring support would be amazing.
Will also check our your shopping list unifier

**Reply by mtojay** [reply]
Bring support would be really really useful for our household. Wife swears on it, adding items to Bring lists is super easy for everyone involved. I showed her mealie and she absolutely loves it and has put in some recipes already - she is sold. But she refuses to let the ease of use of Bring go at this point.

**Comment by HydrelioxGitHub** [comment]
Is this still a feature request or is it on any roadmap ? Love the idea of Bring! and Smart display. Nest Hub have a "recipe mode", could mealie provide data on it ?

**Reply by boc-the-git** [reply]
There isn't a roadmap.
@michael-genson is contemplating this though.

**Comment by boc-the-git** [comment]
Does Home Assistant's [bring integration](https://www.home-assistant.io/integrations/bring) enable most of what this is asking for?

**Comment by michael-genson** [comment]
Bring will be supported with https://github.com/mealie-recipes/mealie/pull/3448. Other integrations are supported through the API and Notifier service.

**Reply by HydrelioxGitHub** [reply]
Wow great ! Can you provide a link to documentation or example on how to deal with other integration using API and Notifier ?

**Reply by michael-genson** [reply]
As a part of PR https://github.com/mealie-recipes/mealie/pull/3448 I added some basic documentation for those features. Aside from recipe actions, all of those features already exist, so you can check them out on your current instance. Until that PR is merged, you can check out the docs I wrote [on my branch here](https://github.com/michael-genson/mealie/blob/feat/recipe-actions/docs/docs/documentation/getting-started/features.md#integrations). I'm happy to provide more info (and update those docs) if you have any questions!

**Comment by michael-genson** [comment]
https://github.com/mealie-recipes/mealie/pull/3448 has been merged into nightly. For more information on how to integrate with Bring! shopping list and other external services, [please check out the Mealie docs](https://docs.mealie.io/documentation/getting-started/features/#recipe-actions).

**Comment by crispybegs** [comment]
I have no idea if it's possible to integrate google keep, but I use it for a shopping list as it's easy to just say to my google nest hubs "add milk to my shopping list" and so my list is updated on every device i own.

**Comment by ecto1a2003** [comment]
This would be great for me as well. Wife uses google keep for grocery shopping and id like it all in one spot
